client/src/main/java/hudson/plugins/swarm/Client.java

```diff
@@ -49,13 +51,27 @@ public static void main(String... args) throws InterruptedException {
             System.exit(0);
         }
 
+        if (options.config != null) {
+            logger.log(Level.INFO, "Load configuration from {0}", options.config.getPath());
+
+            try (InputStream is = new FileInputStream(options.config)) {
+                options = new YamlConfig().loadOptions(is);
+            } catch (IOException | ConfigurationException ex) {
+                System.err.println(ex.getMessage());
+                System.exit(1);
+            }
+        }
+
         validateOptions(options);
 
         // Pass the command line arguments along so that the LabelFileWatcher thread can have them.
         run(new SwarmClient(options), options, args);
     }
 
     private static void validateOptions(Options options) {
+        if (options.url == null) {
+            throw new RuntimeException("Missing 'url' option.");
+        }
```

Comment:
[nit] Failing to provide a URL didn't result in a stack trace before, but now it does. It would be a bit friendlier to keep the old behavior.

Reply:
No problem, can we treat all exceptions thrown in `validateOptions()` like that (catch, print messeage, exit with error)?

Reply:
Sure, that sounds great.